frontend/src/app/containers/ExperimentPage.js

```diff
@@ -648,20 +649,57 @@ export class ExperimentDetail extends React.Component {
     // Ignore subsequent clicks if already in progress
     if (isEnabling) { return; }
 
+    let installAddonPromise = null;
+    if (!this.props.hasAddon) {
+      const { installAddon, requireRestart, eventCategory } = this.props;
+
+      installAddonPromise = installAddon(
+        requireRestart,
+        sendToGA,
+        eventCategory);
+    }
+    let progressButtonWidth;
+    if (this.props.hasAddon) {
+      progressButtonWidth = evt.target.offsetWidth;
+    }
+
     this.setState({
       isEnabling: true,
       isDisabling: false,
       shouldShowTourDialog: true,
-      progressButtonWidth: evt.target.offsetWidth
+      progressButtonWidth
     });
 
-    enableExperiment(experiment);
+    function finishEnabling() {
+      enableExperiment(experiment);
 
-    sendToGA('event', {
-      eventCategory: 'ExperimentDetailsPage Interactions',
-      eventAction: 'Enable Experiment',
-      eventLabel: experiment.title
-    });
+      sendToGA('event', {
+        eventCategory: 'ExperimentDetailsPage Interactions',
+        eventAction: 'Enable Experiment',
+        eventLabel: experiment.title
+      });
+    }
+
+    if (installAddonPromise === null) {
```

Comment:
I wonder if setting `installAddonPromise` to `Promise.resolve()` by default and removing this would be cleaner? It would force the test loop below to run though.

meh. its fine either way